src/networking.c

```diff
@@ -972,30 +974,38 @@ void AddReplyFromClient(client *dst, client *src) {
     closeClientOnOutputBufferLimitReached(dst, 1);
 }
 
-/* Copy 'src' client output buffers into 'dst' client output buffers.
- * The function takes care of freeing the old output buffers of the
- * destination client. */
-void copyClientOutputBuffer(client *dst, client *src) {
-    listEmpty(dst->reply);
-    dst->sentlen = 0;
-    dst->bufpos = 0;
-    dst->reply_bytes = 0;
+/* Logically copy 'src' replica client buffers info to 'dst' replica.
+ * Basically increase referenced buffer block node reference count. */
+void copyReplicaOutputBuffer(client *dst, client *src) {
+    serverAssert(src->bufpos == 0 && listLength(src->reply) == 0);
```

Comment:
i think we needn't always add comments around all this assert. you could see this PR has more than one hundred conversation,  it is not easy to track more useful problem.
please forgive i couldn't follow your code style, i prefer to follow redis original style, every one has his code style, even i also don't like some original styles, maybe before style is not good, but i like to follow. I don't want to put energy in this things unless core team ask me to change, for style problems, i would like to respect the decision of core team just because they always are responsible for this project.